pennylane/wires.py

```diff
@@ -493,6 +493,186 @@ def unique_wires(list_of_wires):
 
         return Wires(tuple(unique), _override=True)
 
+    def union(self, other):
+        """Return the union of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1.union(wires2)
+        Wires([1, 2, 3, 4, 5])
+        """
+        return Wires(set(self.labels) | set(other.labels))
+
+    def __or__(self, other):
+        """Return the union of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the union with.
+
+        Returns:
+            Wires: A new Wires object representing the union of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
+
+        **Example**
+
+        >>> wires1 = Wires([1, 2, 3])
+        >>> wires2 = Wires([3, 4, 5])
+        >>> wires1 | wires2
+        Wires([1, 2, 3, 4, 5])
+        """
+        if not isinstance(other, Wires):
+            raise TypeError(f"Can only do the union of Wires with Wires. Got {type(other)}.")
+        return self.union(other)
+
+    def intersection(self, other):
+        """Return the intersection of the current Wires object and another Wires object.
+
+        Args:
+            other (Wires): Another Wires object to perform the intersection with.
+
+        Returns:
+            Wires: A new Wires object representing the intersection of the two Wires objects.
+
+        Raises:
+            TypeError: If `other` is not an instance of Wires.
```

Comment:
Just a thought... but do we want this? 🤔 What if `other` is another `sequence`, like what `Wires` inherits from? cc @trbromley  

Although I truly don't mind the fact that we're driving users to use `qml.Wires` more than they otherwise would've needed to up until now.

Reply:
+1 for duck typing here

I can imagine users wanting to do the intersection with a list of wires.

Reply:
Ideally we'd like to support `[0, 1, 2] | Wires([3])` as well, which involves defining `__ror__`

Reply:
@austingmhuang sorry we're getting you to go back to the grindstone here hahaha. So, `other` should be allowed to be a `list` as well. This would make `[0, 1, 2] | Wires([3])` possible, as @trbromley suggested.